src/app/space/create/codebases/services/workspaces.service.ts

```diff
@@ -65,7 +67,24 @@ export class WorkspacesService {
         return response.json().data as Workspace[];
       })
       .catch((error) => {
-        return this.handleError(error);
+        if (error.status === 404) {
+          return this.http
+            .get(old_url, {headers: this.headers})
+            .retryWhen(attempts => {
+              let count = 0;
+              return attempts.flatMap(error => {
+                return ++count >= 10 ? Observable.throw(error) : Observable.timer(count * 3000); // Wait for Che to start
```

Comment:
@joshuawilson We retry 10 times with a 3s pause. 
@ibuziuk is there an endpoint for `status`? so we wouldn't have to do those long retries.

Reply:
@corinnekrych yeah, here is the state endpoint[1]. Also, imo there is no need to retry if the error is 404 (When che server is not ready status code would be 503 / 504 ). Basically, if we got 404 than we should try new endpoint without checking state


[1] https://github.com/fabric8-services/fabric8-wit/blob/master/design/codebases.go#L214